mpi-proxy-split/mpi-wrappers/mpi_request_wrappers.cpp

```diff
@@ -393,6 +396,38 @@ USER_DEFINED_WRAPPER(int, Request_get_status, (MPI_Request) request,
   return retval;
 }
 
+USER_DEFINED_WRAPPER(int, Cancel, (MPI_Request *) request)
+{
+  int retval;
+  DMTCP_PLUGIN_DISABLE_CKPT();
+  MPI_Request realRequest = VIRTUAL_TO_REAL_REQUEST(*request);
+  JUMP_TO_LOWER_HALF(lh_info.fsaddr);
+  if( realRequest == MPI_REQUEST_NULL ){
+    retval = MPI_SUCCESS;
+  } else {
+    retval = NEXT_FUNC(Cancel)(&realRequest);
+  }
+  RETURN_TO_UPPER_HALF();
+  DMTCP_PLUGIN_ENABLE_CKPT();
+  return retval;
+}
+
+USER_DEFINED_WRAPPER(int, Request_free, (MPI_Request *) request)
+{
+  int retval;
+  DMTCP_PLUGIN_DISABLE_CKPT();
+  MPI_Request realRequest = VIRTUAL_TO_REAL_REQUEST(*request);
+  JUMP_TO_LOWER_HALF(lh_info.fsaddr);
+  if( realRequest == MPI_REQUEST_NULL ){
```

Comment:
Same question here, as above.  If the lower half will handle `MPI_REQUEST_NULL` correctly, then why do we need to duplicate that logic in MANA?

Reply:
Good afternoon!
You are right. Truth is I encountered a problem and this if-statement seemed to solve it, but I have reverted to not having the if-statement and it works just as good!